### PR TITLE
Feature/94 date types

### PR DESCRIPTION
**Pull request by adprus**
#94: implemented 3rd party date type check

### CODE REVIEW

**Review by lmarniazman — CHANGES_REQUESTED** [review]
Great work, thank you.

Functionality-wise it works, just be careful to use devonfw format (e.g. no more than 120 characters per line, indentation of 2 spaces, etc.). 

I'm not sure if 'Blocker' is the right priority for this though. I think setting it to 'Minor' would be better.

Also, could you add this rule into the devon4j.xml so it is included in the quality profile as well?

**Comment by adprus** [comment]
Ok, thanks for your feedback.
I will follow your suggestions.

File: src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyDateTypeCheck.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.sonar.check.Rule;
+
+/**
+ * {@link DevonArchitecture3rdPartyCheck} verifying that proper date types are used.
+ */
+@Rule(key = "E7", name = "devonfw 3rd Party Date Type Check", //
+    priority = Priority.MINOR, tags = { "architecture-violation", "devonfw", "thirdparty" })
+public class DevonArchitecture3rdPartyDateTypeCheck extends DevonArchitecture3rdPartyCheck {
+
+    @Override protected String checkDependency(JavaType source, JavaType target) {
```
Comment (lmarniazman) [inline src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyDateTypeCheck.java:15]: Annotations (like @Override) should be placed one line above its method / class.

Also the indentation is still at 4 spaces. I am guessing that you set your indentation size to 2 spaces, but didn't set the 'Tab policy' to 'Spaces only'. GitHub only recognizes that there is a tab at the start of the line and automatically parses it to 4 spaces.
Easiest would be to just use the devonfw-ide or import the devonfw formatter config into your preferred IDE.

**Review by lmarniazman — CHANGES_REQUESTED** [review]

File: src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyDateTypeCheck.java (hunk 1 above)
Comment (adprus) [reply src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyDateTypeCheck.java:15]: Yes, I used this devonfw formatter in intellij, but something didn't work, I will check this.

**Comment by adprus** [comment]
#94: reformatted code

**Review by lmarniazman — APPROVED** [review]
Looks good now, thanks!

File: src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyDateTypeCheck.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    priority = Priority.MINOR, tags = { "architecture-violation", "devonfw", "thirdparty" })
+public class DevonArchitecture3rdPartyDateTypeCheck extends DevonArchitecture3rdPartyCheck {
+
+  @Override
+  protected String checkDependency(JavaType source, JavaType target) {
+
+    String targetString = target.toString();
+
+    if (targetString.equals("java.util.Date")) {
+      return "Use the java.time.LocalDate[Time] instead";
```
Comment (hohwille) [inline src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyDateTypeCheck.java:22]: can you remove `the` from the warning message (like in the other cases)?

**Review by hohwille — CHANGES_REQUESTED** [review]
@adprus I added a very small request to change in the review comments. However, just a nice-tohave improvement to your great work. Thanks for the PR 👍

**Comment by hohwille** [comment]
Can you also merge and resolve conflicts please?

**Comment by adprus** [comment]
@hohwille done. Conflicts resolved.

**Pull request merged**